src/tls/openssl/sni.c

```diff
@@ -165,11 +165,15 @@ static int ssl_servername_handler(SSL *ssl, int *al, void *arg)
 	struct tls_cert *uc = NULL;
 	const char *sni;
 
+#if !defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
can you describe which function/symbol is missing here for libressl? I would like to keep ifdefs and branches for libressl and openssl as minimal as possible.

Reply:
Do you mean adding a comment stating why this is necessary?

libressl does not have `TLS_ST_CR_SRVR_HELLO`. I did not find a simple way to determine whether we are in the ClientHello or ServerHello state here with libressl.

Reply:
Is this check really needed? Or is SSL_get_servername enough?

Reply:
For the case when the server requests a certificate from the client, I wanted to skip all the SNI stuff since it doesn't really make sense. But we can probably just remove that check.

Reply:
What do you think? Should I remove it?

Reply:
Yes I would like to keep this simple and without a special openssl branch here. Better to keep this generic. Projects like nginx avoid this too.

Reply:
Sounds good. I found a way that works for both OpenSSL and LibreSSL.